app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt

```diff
@@ -44,14 +45,29 @@ import com.github.se.icebreakrr.ui.sections.shared.TopBar
 import com.google.firebase.Firebase
 import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.auth.auth
+import kotlinx.coroutines.launch
 
 // This File was written with the help of Cursor
 @Composable
-fun SettingsScreen(profilesViewModel: ProfilesViewModel, navigationActions: NavigationActions) {
+fun SettingsScreen(
+    profilesViewModel: ProfilesViewModel,
+    navigationActions: NavigationActions,
+    appDataStore: AppDataStore
+) {
   val context = LocalContext.current
   val scrollState = rememberScrollState()
+  val coroutineScope = rememberCoroutineScope()
+
+  // Collect the discoverability state from DataStore
+  val isDiscoverable by appDataStore.isDiscoverable.collectAsState(initial = true)
+
   lateinit var auth: FirebaseAuth
 
+  // Constants for padding and spacing
+  val screenPadding = 16.dp
+  val verticalSpacing = 8.dp
+  val buttonVerticalSpacing = 16.dp
```

Comment:
Nitpick: If they are constants, use const val and write the names in uppercase.

Reply:
I believe in the meantime Maximo did a PR fixing the magic numbers in these 2 screens, I'll merge and accept his when he merges with the main. But you are right.